ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/BlockPropagationManager.java

```diff
@@ -105,9 +110,34 @@ private void setupListeners() {
         .subscribe(EthPV62.NEW_BLOCK_HASHES, this::handleNewBlockHashesFromNetwork);
   }
 
+  @VisibleForTesting
+  void broadcastBlock(final Block block) {
```

Comment:
Since the validation is being done in here this should probably be called `validateAndBroadcastBlock`.  And hopefully we don't need it to be `@VisibleForTesting` - it's a code smell that tests need access to private internals of the class being tested.